File: SHCardView/SHCardView/ViewController.m
//
//  ViewController.m
//  SHCardView
//
//  Created by shihao on 2017/9/19.
//  Copyright © 2017年 shihao. All rights reserved.
//

#import "ViewController.h"
#import "SHCardView.h"
#import "SHTestCardView.h"

@interface ViewController ()<SHCardViewDelegate,SHCardViewDataSource>
@property (strong, nonatomic) IBOutlet SHCardView *cardView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    UIWindow *window = [UIApplication sharedApplication].keyWindow;
    
    [window addSubview:self.cardView];
    self.cardView.datasource = self;
    self.cardView.delegate = self;
    
    self.cardView.translatesAutoresizingMaskIntoConstraints = NO;
    [self.cardView.leadingAnchor constraintEqualToAnchor:window.leadingAnchor].active = YES;
    [self.cardView.trailingAnchor constraintEqualToAnchor:window.trailingAnchor].active = YES;
    [self.cardView.bottomAnchor constraintEqualToAnchor:window.bottomAnchor].active = YES;
    [self.cardView.topAnchor constraintEqualToAnchor:window.topAnchor constant:64].active = YES;
}

- (void)viewWillAppear:(BOOL)animated{
    [super viewWillAppear:animated];
    [self.cardView reloadData];
}

- (IBAction)reloadBarItemClick:(id)sender {
    [self.cardView reloadData];
}

#pragma mark - <SHCardViewDataSource>

- (NSUInteger)numberOfCardsInCardView:(SHCardView *)cardView
{
    return 10;
}

- (UIView *)cardView:(SHCardView *)cardView viewForCardAtIndex:(NSUInteger)index
{
    NSLog(@"%@ - %@", @(__FUNCTION__), @(index));
    
    SHTestCardView *card = [cardView dequeueReusrIdentifier:@"SHTestCardView"];
    if (!card) {
        card = [SHTestCardView testCardView];
        card.frame = ({
            CGSize screenSize = [UIScreen mainScreen].bounds.size;
            CGRect frame = card.frame;
            frame.size = CGSizeMake(screenSize.width - 40, screenSize.height - 64 - 80);
            frame;
        });
        __weak SHCardView *weakCardView = cardView;
        card.likeAction = ^{
            [weakCardView throwTopCardOnDirection:SHCardViewDirectionTop angle:M_PI_2 / 2];
        };
        card.disLikeAction = ^{
            [weakCardView throwTopCardOnDirection:SHCardViewDirectionBottom angle:M_PI_2 * 3 / 2];
        };
    }
    
    card.indexLabel.text = [NSString stringWithFormat:@"%lu", index];
    
    return card;
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: ProgressAnimation-Swift/ProgressAnimation-Swift/ViewController.swift
//
//  ViewController.swift
//  ProgressAnimation-Swift
//
//  Created by shihao on 2017/8/31.
//  Copyright © 2017年 shihao. All rights reserved.
//

import UIKit

class ViewController: UIViewController {


    @IBOutlet weak var progressView: Progress!
    @IBOutlet weak var percentlabel: UILabel!
    
    let total : CGFloat = 128
    let currentRange: CGFloat = 0
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        progressView.curValue = CGFloat(currentRange)
        progressView.range = CGFloat(total)
    }


    @IBAction func incrementProgress(_ sender: UIButton) {
        
        guard progressView.curValue < progressView.range else {
            return
        }
        
        let incrementRange = 8
        progressView.curValue = progressView.curValue + CGFloat(incrementRange)
        
        //label content
        let percent = progressView.curValue / progressView.range
        self.percentlabel.text = numberFromPercentage(Double(percent))
    }
    
    func numberFromPercentage(_ number : Double) -> String {
        
        let numberformat = NumberFormatter()
        numberformat.numberStyle = .percent
        numberformat.percentSymbol = ""
        return numberformat.string(from: NSNumber(value:number))!
    }

}

